setup.py

```diff
@@ -99,7 +106,7 @@ def numpy_build_ext(pars):
     class build_ext(_build_ext):
         def finalize_options(self):
             _build_ext.finalize_options(self)
-            __builtins__.__NUMPY_SETUP__ = False
+            set_builtin('__NUMPY_SETUP__', False)
```

Comment:
@lwgordonimos Would you mind making this a local function to the `build_ext.finalize_options` method? i.e.
```python
class build_ext(_build_ext):
    def finalize_options(self):
        def _set_builtin(name, value):
            if isinstance(__builtins__, dict):
                __builtins__[name] = value
            else:
                setattr(__builtins__, name, value)

        _build_ext.finalize_options(self)
        _set_builtin('__NUMPY_SETUP__', False)
        import numpy
        self.include_dirs.append(numpy.get_include())
```
as it's only really relevant to the `build_ext.finalize_options` method, thanks.

Reply:
Not at all, done!

Reply:
@lwgordonimos Nice one, thanks... but you've made the `set_builtin` function local to the `numpy_build_ext` factory function, rather than the `finalize_options` method in the `build_ext` class, see my example above :wink: 

I'm keen to keep the `set_builtin` function tightly coupled to where it is used i.e. `build_ext.finalize_options`

Would you be happy to do that? Much obliged :smile: